js/app_credential_product_list.js

```diff
@@ -0,0 +1,42 @@
+"use strict";
+
+Drupal.behaviors.appCredentialProductList = {
+  isSetUp: false,
+  attach: function (context, drupalSettings) {
+    if (this.isSetUp) {
+      return;
+    }
+
+    var rows = document.querySelectorAll('.api-product-list-row');
+    if (rows.length < drupalSettings.numOfVisibleRows) {
+      return;
+    }
+
+    var hidden = 'app-credential-product-list--hidden';
+
+    // NodeList to Array.
+    var rowsToHide = Array.prototype.slice.call(rows).slice(drupalSettings.numOfVisibleRows);
```

Comment:
Wouldn't be better to create this markup already in the Twig file? So the JS would only hide/show the products and the buttons.